### PR TITLE
add register multiple components and test cases

### PR DESCRIPTION
**Pull request by devinsag**
please see https://gitlab.com/architect-io/architect-cli/-/issues/470

### CODE REVIEW

File: test/commands/register.test.ts [hunk 1]
```diff
@@ -410,4 +410,136 @@ describe('register', function () {
...
+      .reply(200, mock_account_response)
+    )
+    .nock(MOCK_API_HOST, api => api
+      .persist()
+      .post(/\/accounts\/.*\/components/)
+      .reply(200, {})
+    )
+    .stdout({ print })
+    .stderr({ print })
+    .command(['register', '-t', '1.0.0', '-a', 'examples', 'examples/react-app/architect.yml', 'examples/gcp-pubsub/pubsub/architect.yml'])
```
Comment (ryan-cahill) [inline test/commands/register.test.ts:457]: I think that this is the most intuitive way to still be able to use the `-t` flag in combination with multiple components being registered, though I do think that we should mention how this works in the `-t` flag description

**Review by ryan-cahill — CHANGES_REQUESTED** [review]

File: test/commands/register.test.ts [hunk 2]
```diff
@@ -410,4 +410,136 @@ describe('register', function () {
...
+    .stdout({ print })
+    .stderr({ print })
+    .command(['register', 'examples/react-app/architect.yml', '-t', '1.0.0', 'examples/gcp-pubsub/pubsub/architect.yml', '--arg', 'NODE_ENV=dev', '-a', 'examples' ])
+    .it('register multiple apps at the same time with a shared build arg', ctx => {
+      expect(ctx.stderr).to.contain('Registering component react-app:1.0.0 with Architect Cloud...... done\n');
+      expect(ctx.stderr).to.contain('Registering component examples/gcp-pubsub:1.0.0 with Architect Cloud...... done\n');
+      expect(ctx.stdout).to.contain('Successfully registered component');
+      const compose = DockerBuildXUtils.dockerBuildX as sinon.SinonStub;
+      expect(compose.callCount).to.eq(2);
+      expect(compose.firstCall.args[0][5]).to.deep.equal("*.args.NODE_ENV=dev");
```
[the remark this answers is not in the document]
Comment (devinsag) [reply test/commands/register.test.ts:542]: good catch, thank you. Added an expect statement to check `secondCall` as well.

File: test/commands/register.test.ts [hunk 3]
```diff
@@ -410,4 +410,136 @@ describe('register', function () {
...
+    )
+    .nock(MOCK_API_HOST, api => api
+      .persist()
+      .post(/\/accounts\/.*\/components/)
+      .reply(200, {})
+    )
+    .stdout({ print })
+    .stderr({ print })
+    .command(['register', 'examples/react-app/architect.yml', '-t', '1.0.0', 'examples/gcp-pubsub/pubsub/architect.yml', '--arg', 'NODE_ENV=dev', '-a', 'examples' ])
+    .it('register multiple apps at the same time with a shared build arg', ctx => {
```
[the remark this answers is not in the document]
Comment (devinsag) [reply test/commands/register.test.ts:536]: Added to arg description explaining the behavior of flags when multiple paths are provided.

File: src/commands/register.ts [hunk 4]
```diff
@@ -74,21 +74,41 @@ export default class ComponentRegister extends BaseCommand {
...
+    parsed.args.component = parsed.argv;
+
+    return parsed;
+  }
+
   async run(): Promise<void> {
-    const { flags, args } = await this.parse(ComponentRegister);
+    const { args, flags } = await this.parse(ComponentRegister);
     await Docker.verify();
+    for (const component of args.component) {
```
[the remark this answers is not in the document]
Comment (devinsag) [reply src/commands/register.ts:103]: My concern is the relative pathing shenanigans, although I suppose worst case scenario is they register the same component twice and it takes more time, but I think most users wouldn't hit this case unintentionally. I'll add a few lines to the register.ts file to deduplicate paths and a test to verify `./component_name/architect.yml !== ./component_name/../component_name/architect.yml`.

File: test/commands/register.test.ts (hunk 1 above)
Comment (devinsag) [reply test/commands/register.test.ts:457]: Added description to the -t flag description

File: src/commands/register.ts [hunk 5]
```diff
@@ -74,21 +74,42 @@ export default class ComponentRegister extends BaseCommand {
...
+  }>(options?: Interfaces.Input<F>, argv = this.argv): Promise<Interfaces.ParserOutput<F, A>> {
+    if (!options) {
+      return super.parse(options, argv);
+    }
+    options.args = [];
+    for (const _ of argv) {
+      options.args.push({ name: 'filler' });
+    }
+    const parsed = await super.parse(options, argv) as Interfaces.ParserOutput<F, A>;
+    const relative_component_path_argv = (parsed.argv || []).map(argv => path.resolve(argv));
```
Comment (ryan-cahill) [inline src/commands/register.ts:93]: [`path.resolve`](https://nodejs.org/api/path.html#pathresolvepaths) will return absolute paths, not relative paths

**Review by ryan-cahill — APPROVED** [review]

File: src/commands/register.ts [hunk 6]
```diff
@@ -33,23 +33,23 @@ export default class ComponentRegister extends BaseCommand {
     arg: {
       non_sensitive: true,
```
Comment (tjhiggins) [inline src/commands/register.ts:34]: Need to pull in rc - I renamed non_sensitive to sensitive and moved into the Flag def. See other cmds after updating.

**Review by tjhiggins — CHANGES_REQUESTED** [review]

File: src/commands/register.ts [hunk 7]
```diff
@@ -31,19 +31,19 @@ export default class ComponentRegister extends BaseCommand {
     ...BaseCommand.flags,
     ...AccountUtils.flags,
     arg: Flags.string({
-      description: 'Build arg(s) to pass to docker build',
+      description: 'Build arg(s) to pass to docker build. If multiple components are specified, the same build arg(s) will be applied to each component.',
```
[the remark this answers is not in the document]
Comment (devinsag) [reply src/commands/register.ts]: updated instruction wording to occur in the command description instead of each arg.

**Pull request merged**